internal/serialization/portable_field.go

```diff
@@ -0,0 +1,380 @@
+		sr, err := MarshalOffsetDateTime(f.Value)
+		if err != nil {
+			return ValueNotDecoded
+		} else if sr == nil {
+			return ValueNil
+		} else {
+			return *sr
+		}
+	case PortableTypeTimestampWithTimezoneArray:
+		return ValueNotDecoded
+	default:
+		return ValueUnknown
+	}
+	return v
+}
+
+// PortableFieldType corresponds to FieldDefinitionType+1
+type PortableFieldType int32
+
+const (
+	PortableTypeNone                       PortableFieldType = 0
+	PortableTypePortable                   PortableFieldType = 1
+	PortableTypeByte                       PortableFieldType = 2
+	PortableTypeBool                       PortableFieldType = 3
+	PortableTypeUint16                     PortableFieldType = 4
+	PortableTypeInt16                      PortableFieldType = 5
+	PortableTypeInt32                      PortableFieldType = 6
+	PortableTypeInt64                      PortableFieldType = 7
+	PortableTypeFloat32                    PortableFieldType = 8
+	PortableTypeFloat64                    PortableFieldType = 9
+	PortableTypeString                     PortableFieldType = 10
+	PortableTypePortableArray              PortableFieldType = 11
+	PortableTypeByteArray                  PortableFieldType = 12
+	PortableTypeBoolArray                  PortableFieldType = 13
+	PortableTypeUInt16Array                PortableFieldType = 14
+	PortableTypeInt16Array                 PortableFieldType = 15
+	PortableTypeInt32Array                 PortableFieldType = 16
+	PortableTypeInt64Array                 PortableFieldType = 17
+	PortableTypeFloat32Array               PortableFieldType = 18
+	PortableTypeFloat64Array               PortableFieldType = 19
+	PortableTypeStringArray                PortableFieldType = 20
+	PortableTypeDecimal                    PortableFieldType = 21
+	PortableTypeDecimalArray               PortableFieldType = 22
+	PortableTypeTime                       PortableFieldType = 23
+	PortableTypeTimeArray                  PortableFieldType = 24
+	PortableTypeDate                       PortableFieldType = 25
+	PortableTypeDateArray                  PortableFieldType = 26
+	PortableTypeTimestamp                  PortableFieldType = 27
+	PortableTypeTimestampArray             PortableFieldType = 28
+	PortableTypeTimestampWithTimezone      PortableFieldType = 29
+	PortableTypeTimestampWithTimezoneArray PortableFieldType = 30
+)
+
+func (t PortableFieldType) MarshalText() ([]byte, error) {
+	var s string
+	switch t {
+	case PortableTypeNone:
+		s = ""
+	case PortableTypePortable:
+		s = "portable"
+	case PortableTypeByte:
+		s = "byte"
+	case PortableTypeBool:
+		s = "bool"
+	case PortableTypeUint16:
+		s = "uint16"
+	case PortableTypeInt16:
+		s = "int16"
+	case PortableTypeInt32:
+		s = "int32"
+	case PortableTypeInt64:
+		s = "int64"
+	case PortableTypeFloat32:
+		s = "float32"
+	case PortableTypeFloat64:
+		s = "float64"
+	case PortableTypeString:
+		s = "string"
+	case PortableTypePortableArray:
+		s = "portablearray"
```

Comment:
Java client uses uppercase notation for these strings with `_` between words. I think we should follow same here.

Reply:
I guess it's a matter of preference. We can discuss that later, if it's OK with you.